### ISSUE TITLE
dcl(HTMLElement, { constructor: func }) fails on iOS 8, Safari desktop 8

### ISSUE THREAD

**Issue by wkeese** [uhop/dcl#15]
The code below fails on iOS 8 with an error about attempting to set the `constructor` property, which is readonly:

``` js
dcl(HTMLElement, {
   constructor: function () { }
});
```

See ibm-js/delite#259 for original ticket.

The error is raised by this statement in the `buildStubs()` function in dcl/mini.js:

``` js
proto[name] = dcl._sb(weaver[name], bases, name, chains);
```

(https://github.com/uhop/dcl/blob/1.1.1/mini.js#L221 - I point to dcl 1.1.1 which is the release we are using - it is the latest available), the call of buildStubs() being here: https://github.com/uhop/dcl/blob/1.1.1/mini.js#L109 .

The call stack:

```
[Error] Error: Attempted to assign to readonly property.
buildStubs@http://9.128.82.125:8020/dojo2-pure/dcl/mini.js:221:9
dcl@http://9.128.82.125:8020/dojo2-pure/dcl/mini.js:109:13
register@http://9.128.82.125:8020/dojo2-pure/delite/register.js:314:17
simple@http://9.128.82.125:8020/dojo2-pure/delite/tests/unit/register.js:39:25
```

The debugger shows that, when it fails, name is "constructor" and bases is [Object, HTMLElementConstructor].

It sounds similar to #9 which we've worked around by overriding `dcl.mix() to ignore exceptions, but that's not solving the problem in this case because`buildStubs()`has its own`for()` loop.   Also, it's a local function so we can't override it.

**Comment by wkeese** [comment]
Unfortunately this is now happening in Safari desktop (mac) too, version 8.0.4.

**Comment by wkeese** [comment]
You marked this as "ready" a few months ago.  Can you release the code soon?

I also want to test it; I'm worried your change won't be sufficient because the problem sometimes happens even when the user never specifies a property called "constructor", since the DCL code itself creates such a property (see the `cname = "constructor"` code in mini.js).

And of course, the problem is exacerbated since the problem can happen setting "constructor" on any class, not just the root HTMLElement class.

**Comment by wkeese** [comment]
PS: Since "constructor" is a [reserved word](http://www.w3schools.com/js/js_reserved.asp) in javascript you really shouldn't be using it.

**Comment by uhop** [comment]
Actually "constructor" is not a reserved word (no matter what w3school thinks about it, it is not a proper resource): https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords

I fact it is assumed that "constructor" should point to a creator function: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/constructor, and can be changed for non-primitive objects.

Theoretically I can skip it completely &mdash; `dcl` doesn't need it. It is there to be compatible with all "normal classes".

PS: Articles linked above refer to relevant standard documents.

**Comment by wkeese** [comment]
Actually, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/constructor proves my point: that "constructor" is a special word to the language and it's dangerous to try to set it. 

I agree that you should theoretically be able to set the property, and ideally you should be able to, but in reality it's problematic.  Maybe putting it inside a try/catch would solve the problem.

**Comment by wkeese** [comment]
Looks like this is fixed upstream in Safari 10 and iOS 9.3 (presumably in addition to iOS 10).

**Comment by uhop** [comment]
Addressed in 2.0.
